### PR TITLE
Fixes #579

### LINKED ISSUE #579: DefaultRawnessComparer: Unexpected AnnotatedTypeMirror combination
`javac -processor nullness -Awarns `[UnexpectedATMCombination.java](https://gist.github.com/dbrosoft/4313cdbca2eb9cb0990e) throws two runtime exceptions, issuing the following messages:

```
error: DefaultRawnessComparer: Unexpected AnnotatedTypeMirror combination.
  type1 = AnnotatedNullType(  null )
  type2 = AnnotatedTypeVariable( T extends Object )
  visitHistory = VisitHistory( ( T extends Object => capture#876 of ? super T extends Object super T extends Object ), ( Object => Object ) )
```
```
error: DefaultRawnessComparer: Unexpected AnnotatedTypeMirror combination.
  type1 = AnnotatedNullType( null )
  type2 = AnnotatedTypeVariable( T extends @Initialized @Nullable Object )
  visitHistory = VisitHistory( ( T extends @Initialized @Nullable Object => capture#876 of ? super T extends @Initialized @Nullable Object super T extends @Initialized @Nullable Object ), ( @Initialized @Nullable Object => @Initialized @Nullable Object ) )
```

### PR DESCRIPTION
**Pull request by smillst**

### CODE REVIEW

**Comment by smillst** [comment]
This does not actually Fix #579, but rather fixes a different crash.  I'll add code that fixes the example in #579  and reopen this pull request.

File: checker/tests/nullness/Issue579Error.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Addtional Test case for Issue579
```
Comment (wmdietl) [inline checker/tests/nullness/Issue579Error.java]: The "additional" will be clearer in the future if you directly reference "all-systems/Issue579.java"

File: checker/tests/nullness/Issue579Error.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+// https://github.com/typetools/checker-framework/issues/579
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+class Issue579Error{
```
Comment (wmdietl) [inline checker/tests/nullness/Issue579Error.java]: Space after class name.

File: framework/tests/all-systems/Issue579.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// Test case for Issue579
+// https://github.com/typetools/checker-framework/issues/579
+class Issue579{
```
Comment (wmdietl) [inline framework/tests/all-systems/Issue579.java]: Space after class name.

File: framework/tests/all-systems/Issue579.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+    public <T>  void foo(Generic<T> real, Generic<? super T> other, boolean flag) {
+        bar(flag ? real : other);
+    }
+
+    <Q> void bar(Generic<? extends Q> param) {
+    }
+    interface Generic<F> {
```
Comment (wmdietl) [inline framework/tests/all-systems/Issue579.java]: I would also have the empty line as separator here, like in the other test file.

**Pull request closed without merging**